projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/hooks/use-save-to-media-library.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { isBlobURL } from '@wordpress/blob';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useState } from 'react';
+/**
+ * Types
+ */
+import type { BlockEditorStore } from '../../../blocks/ai-assistant/types';
+
+export default function useSaveToMediaLibrary() {
+	const [ isLoading, setIsLoading ] = useState( false );
+	const { getSettings } = useSelect( blockEditorStore, [] ) as BlockEditorStore[ 'selectors' ];
+
+	const saveToMediaLibrary = ( url: string ): Promise< { id: string } > => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const settings = getSettings() as any;
+
+		return new Promise( ( resolve, reject ) => {
+			fetch( url ).then( response => {
```

Comment:
Maybe we should set the loading state here, so it works nicely both when the URL is a base64 image as well as a real remote image file to download.

```suggestion
			setIsLoading( true )
			fetch( url ).then( response => {
```